### PR TITLE
Homescreen Notifications, even though keys were submitted (EXPOSUREAPP-3181, EXPOSUREAPP-3182)

### PR DESCRIPTION
**Pull request by kolyaopahle**
## Description
This PR adds a few additional checks to the test result fetching and risk level calculation tasks to check for prior test submission and stops the notification when tests have been submitted already.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt [hunk 1]
```diff
@@ -615,6 +615,10 @@ object LocalData {
         )
     }
 
+    fun submissionWasSuccessful(): Boolean {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt:618]: val submissionWasSuccessful = submissionWasSuccessful

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt [hunk 2]
```diff
@@ -89,13 +89,18 @@ class DiagnosisTestResultRetrievalPeriodicWorker(
      * @see TestResult
      */
     private fun initiateNotification(testResult: TestResult) {
+        if (LocalData.isTestResultNotificationSent() || LocalData.submissionWasSuccessful()) {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:92]: return if ...

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:92]: afaik this is not possible as return if always requires an else branch

**Review by d4rken — APPROVED** [review]
lgtm

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt (hunk 1 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt]: fixed :)

**Review by harambasicluka — APPROVED** [review]

**Pull request merged**